### PR TITLE
[Playwright] [PEPPER-335] Expand framework to support additional UI

### PR DESCRIPTION
**Pull request by aweng98**
Used Angio study to verify changes; Created new page-objects files; and modified existing function to be more generic.

### CODE REVIEW

File: playwright-e2e/pages/singular/enrollment/about-my-adult-dependent-page.ts [hunk 1]
```diff
@@ -15,8 +14,4 @@ export default class AboutMyAdultDependentPage extends SingularPage {
   firstName(): Input {
     return new Input(this.page, { ddpTestID: 'answer:ABOUT_PATIENT_FIRST_NAME' });
   }
-
-  suggestedAddress(): Card {
-    return new Card(this.page, 'We have checked your address entry and have suggested changes that could help ensure delivery');
-  }
```
Comment (aweng98) [inline playwright-e2e/pages/singular/enrollment/about-my-adult-dependent-page.ts:21]: Not used.

File: playwright-e2e/pages/singular/enrollment/about-my-child-page.ts [hunk 2]
```diff
@@ -27,8 +26,4 @@ export default class AboutMyChildPage extends SingularPage {
   childLastName(): Input {
     return new Input(this.page, { ddpTestID: 'answer:ABOUT_PATIENT_LAST_NAME' });
   }
-
-  suggestedAddress(): Card {
-    return new Card(this.page, 'We have checked your address entry and have suggested changes that could help ensure delivery');
-  }
```
Comment (aweng98) [inline playwright-e2e/pages/singular/enrollment/about-my-child-page.ts:33]: Not used.

File: playwright-e2e/tests/dsm/participant-list/singular-search-participant.spec.ts [hunk 3]
```diff
@@ -22,9 +22,9 @@ test.describe('Singular Study in DSM', () => {
     await participantListPage.waitForReady();
     await participantListPage.openSearchButton().click();
 
-    // Grab one random Short ID from table rwo index: 3 and column index: 3 and use value in the search
+    // Grab one random Short ID from table rwo index:2 and column index:2 and use value in the search
     const table = new Table(page);
-    const shortId = await table.cellLocator(3, 3).innerText();
+    const shortId = await table.cell(2, 2).innerText();
```
Comment (aweng98) [inline playwright-e2e/tests/dsm/participant-list/singular-search-participant.spec.ts:27]: Renamed function, and chagned column and row index to be zero-indexed.

File: playwright-e2e/tests/singular/enrollment/adult-dependent-enrollment.spec.ts [hunk 4]
```diff
@@ -147,15 +147,15 @@ test.describe('Enrol an adult dependent', () => {
     await myDashboardPage.waitForReady();
     const orderedHeaders = ['Title', 'Summary', 'Status', 'Action'];
     const table = myDashboardPage.getDashboardTable();
-    const headers = await table.getColumnNames();
+    const headers = await table.getHeaderNames();
```
Comment (aweng98) [inline playwright-e2e/tests/singular/enrollment/adult-dependent-enrollment.spec.ts:150]: Renamed function.

File: playwright-e2e/lib/widget/table.ts [hunk 5]
```diff
@@ -6,95 +6,96 @@ export default class Table {
...
+    // prettier-ignore
+    this.tableCss = ddpTestID
+        ? `[data-ddp-test="${ddpTestID}"]`
+        : cssClassAttribute
+            ? `table${cssClassAttribute}`
+            : 'table, [role="table"]';
+    this.headerCss = '[role="columnheader"], th[class]';
+    this.rowCss = '[role="row"]:not(mat-header-row), tbody tr';
+    this.cellCss = 'td, [role="cell"]';
+    this.footerCss = 'tfoot tr';
```
Comment (aweng98) [inline playwright-e2e/lib/widget/table.ts:24]: To support `mat-table` and `table` web elements.

**Comment by aweng98** [comment]
@SimoneMaiwald Randomly selected you for review this Playwright PR. Hope you can help. Let me know if you have any question.

**Review by juanseba21 — DISMISSED** [review]

**Comment by aweng98** [comment]
@juanseba21 Sorry for asking, just pushed a new commit to remove ` await this.page.pause()` in `table.ts`. Can you re-approve? Thank you.

File: playwright-e2e/authentication/auth-base.ts [hunk 6]
```diff
@@ -40,5 +40,6 @@ export async function createAccountWithEmailAlias(
   }
   const emailAlias = generateEmailAlias(email);
   await fillInEmailPassword(page, { email: emailAlias, password });
-  return email;
+  console.log(`Created account with email: ${emailAlias}`);
```
Comment (juanseba21) [inline playwright-e2e/authentication/auth-base.ts:43]: Is this console.log() necessary?

Comment (aweng98) [reply playwright-e2e/authentication/auth-base.ts:43]: It's for me, at least. I often run tests on localhost. If I know the email, I can manually log in to check.

**Review by juanseba21 — APPROVED** [review]

**Review by SimoneMaiwald — APPROVED** [review]

**Pull request merged**